Fix panic in elasticsearch input plugin

bobmshannon opened the pull request:
This introduces a fix for #2893. With this change, ES collection is skipped for the interval rather than telegraf crashing due to panic.

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

danielnelson commented on plugins/inputs/elasticsearch/elasticsearch.go:
```diff
@@ -169,7 +169,11 @@ func (e *Elasticsearch) Gather(acc telegraf.Accumulator) error {
-				e.setCatMaster(s + "/_cat/master")
+				if err := e.setCatMaster(s + "/_cat/master"); err != nil {
+					acc.AddError(fmt.Errorf("Unable to retrieve master node information."))
+					acc.AddError(fmt.Errorf(mask.ReplaceAllString(err.Error(), "http(s)://XXX:XXX@")))
```
Since this is a single error, we would need to have a single AddError call, but since it is fatal just return the error.

bobmshannon commented:
@danielnelson `/_cat/master` (and other endpoints, like `/_cluster/health`) will return a `503` in cases where the cluster is unable to elect a master node for some reason. Based on my experience this can be caused by a variety of different things, including but not limited to a misconfiguration usually when first setting up ES, or if the master node(s) have become unresponsive due to some other networking or systems related issue.

But with that being said, there could be other reasons as to why the endpoint would return a non-response that I may be unaware of. ES always returns an error message though, so maybe we could just log that? Something like this:

```
Error in plugin [inputs.elasticsearch]: Unable to retrieve master node information from http(s)://XXX:XXX@/_cat/master. Received status code 503 but expected 200. The error response from server is show below:
{"error":{"root_cause":[{"type":"master_not_discovered_exception","reason":null}],"type":"master_not_discovered_exception","reason":null},"status":503}
```

bobmshannon commented:
@danielnelson I've removed the extra log entry and replaced it with a single one of the following format:
```
elasticsearch: Unable to retrieve master node information. API responded with status-code %d, expected %d
```

which I believe is more in line with the style of the existing error messages.

danielnelson commented:
Will be in 1.3.3 later this week.
